app/src/main/java/com/nextcloud/talk/controllers/ServerSelectionController.kt

```diff
@@ -0,0 +1,373 @@
+import com.nextcloud.talk.R
+import com.nextcloud.talk.api.NcApi
+import com.nextcloud.talk.application.NextcloudTalkApplication
+import com.nextcloud.talk.application.NextcloudTalkApplication.Companion.sharedApplication
+import com.nextcloud.talk.controllers.base.NewBaseController
+import com.nextcloud.talk.controllers.util.viewBinding
+import com.nextcloud.talk.databinding.ControllerServerSelectionBinding
+import com.nextcloud.talk.models.database.UserEntity
+import com.nextcloud.talk.models.json.generic.Status
+import com.nextcloud.talk.utils.AccountUtils.findAccounts
+import com.nextcloud.talk.utils.AccountUtils.getAppNameBasedOnPackage
+import com.nextcloud.talk.utils.ApiUtils
+import com.nextcloud.talk.utils.DisplayUtils
+import com.nextcloud.talk.utils.bundle.BundleKeys.KEY_IS_ACCOUNT_IMPORT
+import com.nextcloud.talk.utils.database.user.UserUtils
+import com.nextcloud.talk.utils.singletons.ApplicationWideMessageHolder
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.Disposable
+import io.reactivex.schedulers.Schedulers
+import java.security.cert.CertificateException
+import javax.inject.Inject
+
+@AutoInjector(NextcloudTalkApplication::class)
+class ServerSelectionController : NewBaseController(R.layout.controller_server_selection) {
+    private val binding: ControllerServerSelectionBinding by viewBinding(ControllerServerSelectionBinding::bind)
+
+    override val appBarLayoutType: AppBarLayoutType
+        get() = AppBarLayoutType.EMPTY
+
+    @Inject
+    lateinit var ncApi: NcApi
+
+    @Inject
+    lateinit var userUtils: UserUtils
+
+    private var statusQueryDisposable: Disposable? = null
+
+    fun onCertClick() {
+        if (activity != null) {
+            KeyChain.choosePrivateKeyAlias(activity!!, { alias: String? ->
+                if (alias != null) {
+                    appPreferences!!.temporaryClientCertAlias = alias
+                } else {
+                    appPreferences!!.removeTemporaryClientCertAlias()
+                }
+                setCertTextView()
+            }, arrayOf("RSA", "EC"), null, null, -1, null)
+        }
+    }
+
+    override fun onViewBound(view: View) {
+        super.onViewBound(view)
+        sharedApplication!!.componentApplication.inject(this)
+        if (activity != null) {
+            activity!!.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_PORTRAIT
+        }
+
+        actionBar?.hide()
+
+        binding.hostUrlInputHelperText.setText(
+            String.format(
+                resources!!.getString(R.string.nc_server_helper_text),
+                resources!!.getString(R.string.nc_server_product_name)
+            )
+        )
+        binding.serverEntryTextInputLayout.setEndIconOnClickListener { checkServerAndProceed() }
+        if (resources!!.getBoolean(R.bool.hide_auth_cert)) {
+            binding.certTextView.visibility = View.GONE
+        }
+        if (resources!!.getBoolean(R.bool.hide_provider) ||
```

Comment:
reading from resources like this seems to fail

Method threw 'android.content.res.Resources$NotFoundException' exception.

Reply:
How did you produce the exception? because it works for me™️

Reply:
ah just tricked myself while debugging. this seems to fail only when evaluating the expression...

![grafik](https://user-images.githubusercontent.com/2932790/154450351-3ed36c6b-e188-4536-90d2-0a5ea3903746.png)